src/convert.js

```diff
@@ -74,7 +74,7 @@ function convertFeature(features, geojson, options, index) {
                 id: id,
                 geometry: geojson.geometry.geometries[i],
                 properties: geojson.properties
-            }, options);
+            }, options, index);
```

Comment:
This should ensure that split geometry collections use the index of the original GeoJSON feature as their id in the auto-generated Id case.